keras/utils/dataset_utils_test.py

```diff
@@ -0,0 +1,357 @@
+    
+    self.assertAllEqual(np.array(list(left_split)).shape,(5,2,32,32,1))
+    self.assertAllEqual(np.array(list(right_split)).shape,(5,2,32,32,1))
+     
+  
+    self.assertAllEqual(list(dataset),list(left_split)+list(right_split)) 
+    
+  # def test_mnist_dataset(self):
+  #   (x_train,y_train),(_,_) = mnist.load_data()
+
+  #   self.assertEqual(x_train.shape,(60000,28,28))
+  #   self.assertEqual(y_train.shape,(60000,))
+    
+  #   dataset = tf.data.Dataset.from_tensor_slices((x_train,y_train))
+    
+  #   with self.assertWarns(ResourceWarning):
+  #     left_split,right_split = dataset_utils.split_dataset(dataset,
+  #                                                          left_size=0.80)
+    
+    
+    
+    
+    
+    
+  # def test_unbatched_tf_dataset_of_dicts_of_vectors(self):
+  #   dataset = tf.data.Dataset.from_tensor_slices(
+  #     {'images': np.random.rand(32,32,3), 
+  #     'labels': np.random.rand(32,1)
+  #     })
+  #   dataset = dataset.batch(2)
+  #   left_split,right_split=dataset_utils.split_dataset(dataset,left_size=5)
+    # self.assertAllEqual(len(list(dataset.unbatch())),len(list(left_split))) 
+
+    
+    
+    
+    
+  def test_list_dataset(self):
+    dataset = [np.ones(shape=(10,10,10)) for _ in range(10)]
+    left_split,right_split = dataset_utils.split_dataset(dataset,
+                                                             left_size=5,
+                                                             right_size=5)
+    self.assertEqual(len(left_split), len(right_split))
+    self.assertIsInstance(left_split, tf.data.Dataset)
+    self.assertIsInstance(left_split, tf.data.Dataset)
+        
+    dataset = [np.ones(shape=(10,10,10)) for _ in range(10)]
+    left_split,right_split = dataset_utils.split_dataset(dataset,
+                                                             left_size=0.6,
+                                                             right_size=0.4)
+    self.assertEqual(len(left_split), 6)
+    self.assertEqual(len(right_split), 4)
+
+    
+  def test_tuple_dataset(self):
+    dataset = (np.ones(shape=(10,10,10)),np.zeros(shape=(10,10,10)))
+    left_split,right_split = dataset_utils.split_dataset(dataset,
+                                                             left_size=0.75,
+                                                             right_size=0.25)
+    self.assertLen(left_split, 8)
+    self.assertLen(right_split, 2)
+    
+    left_split,right_split = dataset_utils.split_dataset(dataset,
+                                                             left_size=0.35,
+                                                             right_size=0.65)
+    self.assertLen(left_split, 4)
+    self.assertLen(right_split, 6)
+    self.assertIsInstance(left_split, tf.data.Dataset)
+    self.assertIsInstance(right_split, tf.data.Dataset)
+    
+  def test_invalid_dataset(self):
+    with self.assertRaises(TypeError):
```

Comment:
Any time you test an error, use `assertRaisesRegex` in order to make sure the exception being raised in the one you were expecting. Otherwise it may be a different one!